doc/nrf/libraries/others/network_core_monitor.rst

```diff
@@ -0,0 +1,122 @@
+.. _network_core_monitor:
+
+Network Core Monitor
+####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Network core status monitoring of the nRF5340 processor.
+
+Overview
+********
+
+The Network Core Monitor module provides network core status monitoring by application core for the nRF5340 processor.
+It allows detection of network processor suspensions and resets.
+After a reset occurs, it is possible to read the cause of the reset.
+
+Implementation
+==============
+
+The module uses two general-purpose registers of the IPC peripheral of the application core.
+It writes the state of the network processor to them.
+
+.. figure:: images/ncm_register.svg
+   :alt: Application of IPC peripheral registers.
+
+The ``GPMEM[0]`` register is divided into two 16bit parts.
+This is shown in the figure above.
+
+The value of the counter - *COUNTER* is incremented by the network core every :kconfig:option:`CONFIG_NCM_APP_FEEDING_INTERVAL_MSEC` Kconfig option in milliseconds.
+The application core periodically calls the :c:func:`ncm_net_status_check` function to check the status of the network core.
+If the network core is suspended, the counter value is not updated. The :c:func:`ncm_net_status_check` function returns the error code ``-EBUSY``.
+
+The :c:func:`ncm_net_status_check` function should be called less frequently than the value of the :kconfig:option:`CONFIG_NCM_APP_FEEDING_INTERVAL_MSEC` Kconfig option.
+
+The *FLAGS* field contains the following entries:
+
+   - Reset - Bit 0
+
+The reset bit is set at the boot of the network core.
+It informs the network core that a reset has occurred.
+
+During startup of the networkore core, in addition to setting the reset bit, the cause of the reset is written to the IPC register ``GPMEM[1]``.
+This value is rewritten from the networkore core's ``RESET.RESETREAS`` register.
+See the documentation for a detailed description of the bits in this register: `nRF5340 Reset reason`_
+
+The :c:func:`ncm_net_status_check` function returns the following error codes:
+
+   - ``-EBUSY`` if the network core is suspended
+   - ``-EFAULT`` if a network core reset has occurred
+
+The function takes as parameter a pointer to a variable of type uin32_t.
+When a network core reset is detected, the cause of the reset is stored in this pointer.
+
+Configuration
+*************
+
+To enable Network Core Monitor, set the :kconfig:option:`CONFIG_NET_CORE_MONITOR=y` Kconfig option on both network and application cores.
+
+The :kconfig:option:`NCM_APP_FEEDING_INTERVAL_MSEC` Kconfig option specifies how often the counter is updated by the network core.
+The default value is 500 milliseconds.
+
+The :kconfig:option:`CONFIG_RESET_INIT_PRIORITY` Kconfig option is a priority run function whose purpose is to read the cause of the processor reset and pass this information to the application core.
+It is executed at the network core boot.
```

Comment:
```suggestion
The :kconfig:option:`CONFIG_RESET_INIT_PRIORITY` Kconfig option sets the priority run function.
The function reads the cause of the processor reset and passes this information to the application core.
It is executed at the network core boot.
```

Reply:
I changed the description of this parameter a bit.